File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.js
// Approach: Two Pointers
// Time complexity: O(n)
// Space complexity: O(1)

/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let l = 0, r = numbers.length - 1;   
    while (l < r) {
        let sum = numbers[l] + numbers[r];
        if (sum > target) {
            r--
        } else if (sum < target) {
            l++
        } else {
            return [l + 1, r + 1]
        }
    } 
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
# Approach 1: One-pass Hash Table => val: index
# Time complexity: O(n)
# Space complexity: O(n)
    
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        hashmap = {}
        for i in range(len(numbers)):
            diff = target - numbers[i]
            if diff in hashmap:
                return [hashmap[diff] + 1, i + 1]
            hashmap[numbers[i]] = i